src/wallet/index.ts

```diff
@@ -69,11 +69,12 @@ const registerAccountChangedListener = () => {
  * @returns address of the connected account
  */
 export const connect = async (): Promise<string | null> => {
```

Comment:
the function `checkIsWalletInstalled` should throw an error.

```
export const checkIsWalletInstalled = () => {
  if (typeof window === "undefined") {
    throw new Error("Should run through Browser");
  }
  // Have to check the ethereum binding on the window object to see if it's installed
  const { ethereum } = window;

  if (ethereum && ethereum?.isMetaMask) return ethereum;

  throw new Error("No wallet installed");
};

```

Reply:
why should a **check-only** function throw an error in your opinion? it's just a yes/no information you get if called... which is exactly what we need to inform the user about a lack of a web3 wallet. but have a look at https://github.com/unicrowio/sdk/pull/118, i think it's better to have it implemented that way.

Reply:
to avoid if/else

Reply:
and handle it properly using try/catch, once the user of SDK should have web3wallet installed.

Reply:
but ok let's keep this way for now, just fix the conflict and lets merge